decompiler/frontend/binaryninja/parser.py

```diff
@@ -46,6 +47,22 @@ def parse(self, function: Function) -> ControlFlowGraph:
         self._report_lifter_errors()
         return cfg
 
+    def _fix_switch_edge_address(self, edge, lookup_table):
+        """
+        If given edge.target.source_block.start address is not in lookup table, inspect addresses used in edge.target.
+        Return matched case list for edge.target.
+        """
+        possible_matches = set()
+        for instruction in edge.target:
+            match instruction:
+                # tail calls destroy edge address mapping
+                case MediumLevelILTailcallSsa(dest=MediumLevelILConstPtr()):
+                    possible_matches.add(instruction.dest.constant)
```

Comment:
Why a match with one case?

Reply:
For one I find the match case syntax elegant, compared to 
```
          if isinstance(instruction, MediumLevelILTailcallSsa) and isinstance(instruction.dest, MediumLevelILConstPtr):
                ...
```
My initial though was that I have to account for additional cases but this did not seem to be necessary...
Should I change it to an if statement?